doc/specs/Configuration object for profiles.md

```diff
@@ -0,0 +1,125 @@
+and easier to understand than the alternative, where each setting within an `unfocusedConfig` object has a parent from which
+it obtains its value.
+
+Note that when `TerminalApp` initializes a control, it creates a `TerminalSettings` object for that profile and passes the
+control a child of that object (so that the control can store overrides in the child, as described earlier). If an unfocused
+config is defined in the profile (or in globals/profile defaults), then `TerminalApp` will create a *child of that child*,
+put the relevant overrides in it, and pass that into the control as well. Thus, the inheritance of any undefined parameters
+in the unfocused config will be as follows:
+
+1. The unfocused config specified in the profile (or in globals/profile defaults)
+2. Overrides made by term control
+3. The parent profile
+
+## UI/UX Design
+
+Users will be able to add a new setting to their profiles that will look like this:
+
+```
+"unfocusedConfig":
+{
+    "colorScheme": "Campbell",
+    "cursorColor": "#888",
+    "cursorShape": "emptyBox",
+    "foreground": "#C0C0C0",
+    "background": "#000000"
+}
+```
+
+When certain appearance settings are changed via OSC sequences (such as the background color), only the focused/regular
+appearance will change and the unfocused one will remain unchanged. However, since the unfocused settings object inherits
+from the regular one, it will still apply the change (provided it does not define its own value for that setting).
+
+## Capabilities
+
+### Accessibility
+
+Does not affect accessibility.
+
+### Security
+
+Does not affect security.
+
+### Reliability
+
+This is another location in the settings where parsing/loading the settings may fail. However, this is the case
+for any new setting we add so I would say that this is a reasonable cost for this feature.
+
+### Compatibility
+
+Should not affect compatibility.
+
+### Performance, Power, and Efficiency
+
+## Potential Issues
+
+Inactive tabs will be 'rendered' in the background with the `UnfocusedRenderingParams` object, we need to make
+sure that switching to an inactive tab (and so causing the renderer to update with the 'normal' parameters)
+does not cause the window to flash/show a jarring indicator that the rendering values changed.
+
+## Future considerations
+
+We will need to decide how this will look in the settings UI.
+
+We may wish to add more states in the future (like 'elevated'). When that happens, we will need to deal with how
+these appearance objects can scale/layer over each other. We had a lot of discussion about this and could not find
+a suitable solution to the problem of multiple states being valid at the same time (like unfocused and elevated).
+This, along with the fact that it is uncertain if there even will be more states we would want to add led us to
+the conclusion that we should only support the unfocused state for now, and come back to this issue later. If there

```

Comment:
Well.... the way the json snippet is... it's a bit awkward to come back later. Unless the json snippet is changed to be like { "stateChangeConfig" : { "mode" : "unfocused", "overrides" : { ... blah blah blah } } }... in which case it becomes future proof and can become "mode" : "elevated" or "mode" : "elevated | unfocused" in the future.

I'm just thinking that if you have to make >1 in the future, it's a little awkward to have a bunch of different formats for different states.

Reply:
We can always deprecate one version of it.